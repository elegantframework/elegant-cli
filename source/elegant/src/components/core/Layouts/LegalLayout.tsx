interface Props {
    /**
     * HTML child elements that can be passed to the component.
     */
    children: React.ReactNode;
};
/**
 * The layout component to be used for legal pages.
 * @returns An outer parent html component to be used for legal pages.
 */
const LegalLayout = ({
    children
}: Props) => {
    return (
        <main className="max-w-3xl mx-auto relative z-20 pt-10 pb-20 sm:pb-10 px-6 sm:px-8 md:px-10">
          {children}
        </main>
    );
};

export default LegalLayout;