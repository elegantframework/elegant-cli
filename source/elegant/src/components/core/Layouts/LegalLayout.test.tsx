import React from "react";
import renderer from 'react-test-renderer';
import LegalLayout from "./LegalLayout";

describe("Legal Page layout component", () => {
    it('renders the proper html as expected', () => {
        const layout = renderer
        .create(
            <LegalLayout>
                <div>
                    Hello World!
                </div>
            </LegalLayout>
        )
        .toJSON();
        expect(layout).toMatchSnapshot();
    });
});