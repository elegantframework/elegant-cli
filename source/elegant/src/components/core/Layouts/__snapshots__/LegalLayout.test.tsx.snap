// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal Page layout component renders the proper html as expected 1`] = `
<main
  className="max-w-3xl mx-auto relative z-20 pt-10 pb-20 sm:pb-10 px-6 sm:px-8 md:px-10"
>
  <div>
    Hello World!
  </div>
</main>
`;
